### DISCUSSION TITLE
GMKtec M7 Ryzen 7 6850H 680m Finally WORKED - Win10 LinuxLite

### DISCUSSION THREAD

**Discussion by ioddevs** [isc30/ryzen-gpu-passthrough-proxmox#80] [Show and tell]
GMKtec M7 Ryzen 7 6850H Finally WORKED
![image](https://github.com/user-attachments/assets/55b2b021-1658-4e2f-b2a4-f0f059dbd2d8)

but then it sunk :D:D:D
still troubleshooting.

whats done:
- rebuild bin
- use default rom
- still in OVMF bios

**Comment by ioddevs** [comment]
ok it worked !!!
the key is.. dont CRASH!!! if crash host reboot needed lol

Proxmox 8.2.7 Running paralel 
- 4x LXC 1core and min RAM 
- 1x Win10 passthrough RTX3060-12G 8core 64G ram <-- right monitor
- 1x Win10 no GPU 2core 8G ram <-- on another laptop RDP
- 1x Win10 passthrough iGPU AMD Radeon 680m 16G Shared VRam 4core 8G ram <-- TV
![IMG_0651](https://github.com/user-attachments/assets/5385f075-7627-434c-b77e-613ff505afa6)

so happy to maxed out very economical mini pc :) 
------------------------------------------------------------
ANNDDDD !!!!
it worked with exact same setting on linux lite, also ovmf
- LinuxLite passthrough iGPU AMD Radeon 680m 16G Shared VRam 2core 8G ram <-- TV

linux seems smoother
![IMG_0656](https://github.com/user-attachments/assets/15452e49-1465-477a-8a8e-24955c6e51fd)

**Comment by ioddevs** [comment]
I noticed actually 
- no need primary GPU
- ovmf works better than seabios
- prep all parameter before install
- display default use vnc, Install then shutdown
- Then change display none, use monitor.
- Some linux like bazzite does not work well with real change , bar disappeared, cannot login, etc
- similar setting works with Nvidia eGPU just easier as no need to add extra bin and rom param

Now I don’t get it.. why it took me a month to get it running like this lol.. the Nvidia one seems so simple

**Comment by Julik99** [comment]
Hi! I have exactly the same computer (GMKtec M7 Ryzen 7 6850H). Please help me set it up. I want to use a virtual machine Windows 10 and Ubuntu. I can't compile the BIOS according to the instructions. I used a ready-made BIOS from 6900hx. But apparently it doesn't work. I did everything according to the instructions, but the images are not transmitted via HDMI. 😢

**Comment by Julik99** [comment]
Do I need to make any additional manipulations with the BIOS settings?

**Reply by ioddevs** [reply]
No need.
But you need to

apt install gcc

That part is actually compiling vbios.c code, so it needs gcc compiler.
The steps are so far consistent for me. 
although I can’t seem to replicate for very old ryzen 2400g with Vega GPU.

I will be on long leave soon so won’t be able to help much until mid Dec. But don’t give up. If I am back and u still can’t make it work we could find way to walkthrough. 😃

**Reply by Julik99** [reply]
I was able to compile the BIOS. I did everything, but after starting the virtual machine, I get an error:
TASK ERROR: device '0000:e5:00.1' assigned more than once. but I don't have this device in other virtual machines.
If I remove the sound card in the virtual machine settings, the system freezes (proxymox)
I will be grateful to you if you can help me in the future. 😊

**Reply by kimuss** [reply]
Do you also use tool EfiRom.exe to extract driver for audio? AMDGopDriver?

**Reply by Julik99** [reply]
> AMDGopDrive

I used AMDGopDrive, but my proxymox hangs when I start the virtual machine. Only resetting the computer helps. 😢

**Reply by Julik99** [reply]
Can you attach your bios and files? I will be grateful to you =)

**Reply by ioddevs** [reply]
hi @Julik99 
how did it go. is it resolved?
i am just partially back from vacation.
bios is the only thing i generate so i attached below.

[vbios_6900HX.zip](https://github.com/user-attachments/files/18224505/vbios_6900HX.zip)

**Comment by Freddyqt** [comment]
I've always had problems, four or six hours after installing Windows. I get a pveproxy error, a swap error, or even an nvme error. I honestly don't know what it is. Has anyone else encountered this error?
